kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/visitor/linux/IfcfgConfigWriter.java

```diff
@@ -123,161 +111,113 @@ private void writeRedhatConfig(NetInterfaceConfig<? extends NetInterfaceAddressC
                 .append(interfaceName).append(".tmp").toString();
         logger.debug("Writing config for {}", interfaceName);
 
-        NetInterfaceType type = netInterfaceConfig.getType();
-        if (type == NetInterfaceType.ETHERNET || type == NetInterfaceType.WIFI || type == NetInterfaceType.LOOPBACK) {
-            StringBuilder sb = new StringBuilder();
-            sb.append("# Networking Interface\n");
-
-            // DEVICE
-            sb.append("DEVICE=").append(netInterfaceConfig.getName()).append("\n");
-
-            // NAME
-            sb.append("NAME=").append(netInterfaceConfig.getName()).append("\n");
-
-            // TYPE
-            sb.append("TYPE=").append(netInterfaceConfig.getType()).append("\n");
-
-            NetInterfaceAddressConfig netInterfaceAddressConfig = ((AbstractNetInterface<?>) netInterfaceConfig)
-                    .getNetInterfaceAddressConfig();
-            boolean allowWrite = false;
-            List<NetConfig> netConfigs = netInterfaceAddressConfig.getConfigs();
-            if (netConfigs != null) {
-                for (NetConfig netConfig : netConfigs) {
-                    if (!(netConfig instanceof NetConfigIP4)) {
-                        continue;
-                    }
-                    // ONBOOT
-                    sb.append("ONBOOT=");
-                    if (((NetConfigIP4) netConfig).isAutoConnect()) {
-                        sb.append("yes");
-                    } else {
-                        sb.append("no");
-                    }
-                    sb.append("\n");
-                    if (((NetConfigIP4) netConfig).getStatus() == NetInterfaceStatus.netIPv4StatusL2Only) {
-                        logger.debug("new config is Layer 2 Only");
-                        sb.append("BOOTPROTO=none\n");
-                    } else if (((NetConfigIP4) netConfig).isDhcp()) {
-                        logger.debug("new config is DHCP");
-                        sb.append("BOOTPROTO=dhcp\n");
-                    } else {
-                        logger.debug("new config is STATIC");
-                        sb.append("BOOTPROTO=static\n");
+        StringBuilder sb = new StringBuilder();
+        sb.append("# Networking Interface\n");
 
-                        // IPADDR
-                        sb.append("IPADDR=").append(((NetConfigIP4) netConfig).getAddress().getHostAddress())
-                                .append("\n");
+        // DEVICE
+        sb.append("DEVICE=").append(netInterfaceConfig.getName()).append("\n");
```

Comment:
Is it possible to re-use the constants defined in IfcfgConfig?

Reply:
done